crates/dbsp/src/circuit/circuit_builder.rs

```diff
@@ -3885,13 +4567,35 @@ where
     }
 
     fn commit(&mut self, base: &StoragePath) -> Result<(), DbspError> {
-        self.operator
-            .commit(base, &self.global_id().persistent_id())
+        self.operator.commit(base, self.persistent_id().as_deref())
     }
 
     fn restore(&mut self, base: &StoragePath) -> Result<(), DbspError> {
-        self.operator
-            .restore(base, &self.global_id().persistent_id())
+        self.operator.restore(base, self.persistent_id().as_deref())
```

Comment:
these functions are repeated multiple times, but I don't know how to avoid that
at least, they are very short